### ISSUE TITLE
Behavioral difference in child metadata deployment when using manifest or type/name vs. using source paths

### ISSUE THREAD

**Issue by SCWells72** [forcedotcom/cli#640]
IC has used traditionally a manifest when deploying via `sfdx force:source:deploy`. If necessary for selective deployments, it will generate a temporary `package.xml` file with the specific metadata elements. I had a user point out that when deploying child metadata types, e.g., `CustomField`, this results in the entire parent metadata object's deployment. In other words, if you have a `package.xml` file like:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Package xmlns="http://soap.sforce.com/2006/04/metadata">
    <types>
        <members>MyObject__c.MyField__c</members>
        <name>CustomField</name>
    </types>
    <version>49.0</version>
</Package>
```

and you deploy using:

```
sfdx force:source:deploy -x path/to/that/package.xml
```

it will instead actually deploy all of `MyObject__c` which is not only inefficient but could also result in the deployment of changes that weren't yet ready.

This seems to also be the case when using `-m CustomField:MyObject__c.MyField__c`.

However, if you instead use `-p path/to/objects/MyObject__c/fields/MyField__c.field-meta.xml`, it will deploy only that single field.

I'm updating IC for this specific scenario--effectively the deployment of one or more specific child metadata files will now use source paths instead of a manifest--but it seems like this inconsistency affects any client of `force:source:deploy` because it introduces inconsistent behavior across what should be synonymous invocations and results in the need for specialized behavior in all callers to yield optimal results.

It's worth pointing out that switching to a source path-only model isn't viable because of command-line length limitations on various host OSes, particularly Windows which traditionally has an 8K maximum length that is easily surpassed when trying to deploy all metadata files for even a moderate project. That's why IC has always used a manifest file for deployment to-date.

**Comment by shetzel** [comment]
The fix for this comes with the other fixes for custom fields, this Thursday.  It'll be a CustomField fix party.  🥳

**Comment by shetzel** [comment]
@SCWells72  This is now fixed with CLI v7.76.1 and salesforcedx plugin v50.1.1.  Please reopen if you're still seeing the problem.
